### ISSUE TITLE
Loki Helm chart upgrade from 5.43.7 to 5.44.* fails: can't evaluate field Values in type interface {}

### ISSUE THREAD

**Issue by shoce** [grafana/loki#12238]
**Describe the bug**
I have a Loki installation in k0s kubernetes cluster, with the helm chart version 5.43.7. Upgrading to helm chart version 5.44.0 or 5.44.1 fails with error message (some newlines added manually for readability):
```
Error: UPGRADE FAILED: 
template: loki/templates/ingress.yaml:38:14: 
executing "loki/templates/ingress.yaml" at <include "loki.ingress.servicePaths" $>: 
error calling include: 
template: loki/templates/_helpers.tpl:530:4: 
executing "loki.ingress.servicePaths" at <include "loki.ingress.singleBinaryServicePaths" .>: 
error calling include: 
template: loki/templates/_helpers.tpl:546:4: 
executing "loki.ingress.singleBinaryServicePaths" 
at <include "loki.ingress.servicePath" (
dict "ctx" . "svcName" "singleBinary" "paths" .Values.ingress.paths.singleBinary
)>: 
error calling include: 
template: loki/templates/_helpers.tpl:570:26: 
executing "loki.ingress.servicePath" at <.Values.loki.server.http_listen_port>: 
can't evaluate field Values in type interface {}
```

the loki.values.yaml used for helm install before and for helm upgrade now:
```
---
loki:

  limits_config:
    retention_period: 10w
    ingestion_rate_mb: 12
    ingestion_burst_size_mb: 24
    per_stream_rate_limit: 12MB
    per_stream_rate_limit_burst: 24MB
    split_queries_by_interval: 24h
    max_query_parallelism: 100

  commonConfig:
    path_prefix: /loki
    replication_factor: 1

  storage:
    type: filesystem
    filesystem:
      chunks_directory: /loki/chunks
      rules_directory: /loki/rules

  # license is needed to enable auth with this option
  auth_enabled: false

  query_scheduler:
    max_outstanding_requests_per_tenant: 4096

  frontend:
    max_outstanding_per_tenant: 4096


singleBinary:
  replicas: 1
  persistence:
    enabled: false
  extraVolumes:
   - name: volume-loki
     hostPath:
       path: /volume/loki/
       type: Directory
  extraVolumeMounts:
   - name: volume-loki
     mountPath: /loki
     readOnly: false


ingress:
  enabled: true
  ingressClassName: nginx
  hosts:
    - loki.happydogs.org
  tls:
    - secretName: ""
      hosts:
      - loki.happydogs.org
  paths:
    singleBinary:
      - /api/prom/push
      - /loki/api/v1/push
      - /api/prom/tail
      - /loki/api/v1/tail
      - /loki/api
      - /api/prom/rules
      - /loki/api/v1/rules
      - /prometheus/api/v1/rules
      - /prometheus/api/v1/alerts
      - /ready
      - /log_level
      - /metrics
      - /config
      - /services

gateway:
  enabled: false

monitoring:
  selfMonitoring:
    enabled: true
    grafanaAgent:
      installOperator: true

```

**To Reproduce**
1. Install Loki helm chart version 5.43.7
2. Run helm upgrade to Loki helm chart version 5.44.0

**Expected behavior**
Expected to have successful helm chart upgrade.

**Environment:**
 - Infrastructure: kubernetes k0s v1.29.1+k0s.1
 - Deployment tool: helm version.BuildInfo{Version:"v3.14.0", GitCommit:"3fc9f4b2638e76f26739cd77c7017139be81d0ea", GitTreeState:"clean", GoVersion:"go1.21.5"}

**Screenshots, Promtail config, or terminal output**

**Comment by smuth4** [comment]
It looks like this was fixed in https://github.com/grafana/loki/pull/12241, which is available in 5.46.0

**Comment by shoce** [comment]
> It looks like this was fixed in #12241, which should be available in 5.44.3, however I just upgraded to the latest (5.46.0) and that fixed it for me.

thanks, i have upgraded to v5.46.0 successfully.
